### ISSUE TITLE
Fix accessibility for field help texts

### ISSUE THREAD

**Issue by verheyenkoen** [ugent-library/biblio-backoffice#1650]
# Enhancement
In some places we have some help text below a field. Users using screen readers are missing this context since they will already have filled out the field before they notice the help text. This can be fixed by applying aria labels.

Either we add an `id` attribute to the help text element and reference that using the ~`aria-describedby`~ [`aria-details`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-details) attribute on the form field element (preferred solution). Or we duplicate the help text in the [`aria-description`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-description) attribute on the form field, but then we'd also need to add the [`aria-hidden="true"` attribute](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-hidden) to the help text so this information is not duplicated for them.

Here's an exhaustive list of fields to address:

- Publications
  - Search
    - [x] Created since facet filter
    - [x] Updated since facet filter
  - Add publication(s)
    - [x] ~Add/Import ... options~ (already wrapped inside `<label>`)
    - [x] Import from WoS (label missing + "Web of Science import instructions" help text)
    - [x] Enter identifier of your publication (2 labels missing + "Identifier import instructions" help text)
    - [x] Import via BibTeX file (label missing + "BibTeX import instructions" help text)
  - Edit publication details
    - [x] Publication type
    - [x] DOI
    - [x] Publication year
    - [x] Number of pages
    - [x] Date of defense (dissertation)
    - [x] Web of Science ID
    - [x] ISSN
    - [x] E-ISSN
    - [x] ISBN
    - [x] E-ISBN
    - [x] PubMed ID
    - [x] Arxiv ID
    - [x] ESCI ID
  - Add project
    - [x] Search project
  - Edit conference details
    - [x] Conference start date
    - [x] Conference end date
  - Edit additional information
    - [x] Keywords
  - Files
    - [x] Upload file field (Which document format or version should I use?)
  - Add/edit file
    - [x] Publication version
    - [x] ~Access level~ (already wrapped inside `<label>`)
    - [x] Access level after embargo
    - [x] Embargo end
  - Edit Librarian tags
    - [x] Librarian tags
- Datasets
  - Search
    - [x] Created since facet filter
    - [x] Updated since facet filter
  - Add dataset
    - [x] ~Add/Import ... options~ (already wrapped inside `<label>`)
    - [x] DOI (label missing + help text)
  - Edit dataset details
    - [x] Persistent identifier type
    - [x] Identifier
    - [x] Publication year
    - [x] Keywords
    - [x] Other license
  - Add project
    - [x] Search project
  - Edit Librarian tags
    - [x] Librarian tags
- Batch
  - [x] Operations

- [x] Additionally, we need to make sure all help texts are translated so they are not duplicated (especially in case we need the alternative solution, using `aria-description`).

## Screenshots
![image](https://github.com/ugent-library/biblio-backoffice/assets/599538/ac3c2a27-03b0-4f5c-97bc-96d09851aab5)

HTML currently generated for this field:
```html
<div class="mb-6 row">
  <label class="col-lg-3 col-form-label mt-2" for="type"
    >Publication type</label
  >
  <div class="col-lg-3">
    <select
      class="form-select form-control"
      name="type"
      id="type"
      hx-get="/publication/01J1VXWR3D7DCJ1F544DCDJ9H1/type/confirm"
      hx-headers='{"If-Match": "01J1W9HBZ274660ZAJ02WMHTPR"}'
      hx-target="#modals"
    >
      <option value="journal_article">Journal article</option>
      <option value="book">Book</option>
      <option value="book_chapter">Book chapter</option>
      <option value="book_editor">Book editor</option>
      <option value="issue_editor">Issue editor</option>
      <option value="conference">Conference</option>
      <option value="dissertation">Dissertation</option>
      <option value="miscellaneous" selected="">Miscellaneous</option></select
    ><small class="form-text text-muted"
      >Changing the publication type might result in data loss</small
    >
  </div>
</div>
```

## Additional context
https://getbootstrap.com/docs/5.0/forms/overview/#form-text

# Automatic testing scenario
Ideally this is tested generically for forms. We'd need to query for `.form-text` instances and see if they are linked correctly to the previous field, either via ~`aria-describedby`~ `aria-details` or duplicated in `aria-description` but then the `.form-text` element should also have the `aria-hidden="true"` attribute.

**Comment by mietcls** [comment]
Tasty stuff!

> Either we add an `id` attribute to the help text element and reference that using the `aria-describedby` attribute on the form field element (preferred solution). Or we duplicate the help text in the [`aria-description`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-description) attribute on the form field, but then we'd also need to add the [`aria-hidden="true"` attribute](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-hidden) to the help text so this information is not duplicated for them.

I'd go for the preferred solution so we don't have redundant code nor extra maintenance (duplicate text is doomed to be forgotten).

**Comment by verheyenkoen** [comment]
> I'd go for the preferred solution so we don't have redundant code nor extra maintenance (duplicate text is doomed to be forgotten).

I just added the alternative in case the preferred solution is not "feasible". In case it's not, it should be a translated text anyway (I'll fix it if not), so it's still a one-place-edit.

**Comment by mietcls** [comment]
The instructions won't be as easy to find with `aria-details`, and they should be discovered immediately as there are no good error-fall-backs (there will be more instructions like this).

- Voice assist does not read the instructions out loud, nor sees them with `aria-details`.
<img width="831" alt="Screenshot 2024-07-09 at 11 35 32" src="https://github.com/ugent-library/biblio-backoffice/assets/1572885/5c84393e-4d1f-4c08-b27a-b58027cc0ed0">

- Whenever you adapt it to `aria-describedby`, it did. The links remained accessible (tried with web of science instructions).
<img width="798" alt="Screenshot 2024-07-09 at 11 30 13" src="https://github.com/ugent-library/biblio-backoffice/assets/1572885/36e433f0-5082-477b-be7d-3b13769377df">
<img width="659" alt="Screenshot 2024-07-09 at 11 42 40" src="https://github.com/ugent-library/biblio-backoffice/assets/1572885/f400912b-27dc-4ffe-bc34-829596620fa0">

What was the reason to go with `aria-details` instead of `aria-describedby`?

If there is an issue with `aria-describedby`, we can provide both and hide one, as proposed here https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-details#:

> Unlike `aria-describedby`, elements referenced by `aria-details` are not used in accessible descriptions and are not turned into a plain string when presented to assistive technology users. If the associated content is not too long and flattening the contents of the referenced element to a simple string of text wouldn't cause loss of information, consider using `aria-describedby` instead. That said, it is valid for an element to have both `aria-details` and a description specified with either `aria-describedby` or `aria-description`.

**Comment by verheyenkoen** [comment]
@mietcls That's weird as the [`aria-describedby`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby#:~:text=The%20aria%2Ddescribedby%20property,using%20aria%2Ddetails.) docs don't mention that really. They even seem to encourage using `aria-details` if the content is not plain text. I guess to be safe, we should duplicate them then by using both the `aria-description`, containing a plain text version of whatever is in the `aria-details` element. Ideally we can generate one from the other, however just stripping any HTML tags won't be good enough I guess, especially if they contain hyperlinks.

**Comment by verheyenkoen** [comment]
@mietcls I suggest we build a generic helper function that strips any HTML from whatever is in the element referenced in `aria-details`, except for `<a>`-tags where we could add the url between parentheses after the element content.

So something like: 
```html
<select
    class="form-select form-control"
    id="identifier_type"
    name="identifier_type"
    aria-details="identifier_type-help">
    <option></option>
    <option value="DOI" selected="">DOI</option>
    <option value="BioStudies">BioStudies</option>
    <option value="EGA">EGA</option>
    <option value="ENA">ENA</option>
    <option value="ENABioProject">ENA BioProject</option>
    <option value="Ensembl">Ensembl</option>
    <option value="Handle">Handle</option>
</select>
<small id="identifier_type-help" class="form-text text-muted">
    More information about <a href="https://onderzoektips.ugent.be/en/tips/00001743/" target="_blank">persistent identifiers</a>.<br />
    <a href="mailto:rdm.support@ugent.be">Contact us</a> if your identifier is not in the list.
</small>
```

... would become:

```html
<select
    class="form-select form-control"
    id="identifier_type"
    name="identifier_type"
    aria-details="identifier_type-help"
    aria-description="More information about persistent identifiers (https://onderzoektips.ugent.be/en/tips/00001743/). Contact us (rdm.support@ugent.be) if your identifier is not in the list.">
    <option></option>
    <option value="DOI" selected="">DOI</option>
    <option value="BioStudies">BioStudies</option>
    <option value="EGA">EGA</option>
    <option value="ENA">ENA</option>
    <option value="ENABioProject">ENA BioProject</option>
    <option value="Ensembl">Ensembl</option>
    <option value="Handle">Handle</option>
</select>
<small id="identifier_type-help" class="form-text text-muted">
    More information about <a href="https://onderzoektips.ugent.be/en/tips/00001743/" target="_blank">persistent identifiers</a>.<br />
    <a href="mailto:rdm.support@ugent.be">Contact us</a> if your identifier is not in the list.
</small>
```
